grammars/ruby.cson

```diff
@@ -293,11 +333,32 @@
       '3':
         'name': 'entity.name.function.ruby'
     'comment': ' the optional name is just to catch the def also without a method-name'
-    'match': '(?x)\n\t\t\t         (?=def\\b)                                                           # an optimization to help Oniguruma fail fast\n\t\t\t         (?<=^|\\s)(def)\\b                                                    # the def keyword\n\t\t\t         ( \\s+                                                               # an optional group of whitespace followed by…\n\t\t\t           ( (?>[a-zA-Z_]\\w*(?>\\.|::))?                                      # a method name prefix\n\t\t\t             (?>[a-zA-Z_]\\w*(?>[?!]|=(?!>))?                                 # the method name\n\t\t\t             |===?|!=|>[>=]?|<=>|<[<=]?|[%&`/\\|]|\\*\\*?|=?~|[-+]@?|\\[]=?) ) )?  # …or an operator method\n\t\t\t        '
+    'match': '''(?x)
+      (?=def\\b)                            # optimization to help Oniguruma fail fast
+      (?<=^|\\s)(def)\\b
+      (
+        \\s+
+        (
+          (?>[a-zA-Z_]\\w*(?>\\.|::))?      # method prefix
+          (?>                               # method name
+            [a-zA-Z_]\\w*(?>[?!]|=(?!>))?
+            |===?|!=|>[>=]?|<=>|<[<=]?|[%&`/\\|]|\\*\\*?|=?~|[-+]@?|\\[]=?
+          )
+        )
+      )?
+    '''
     'name': 'meta.function.method.without-arguments.ruby'
   }
   {
-    'match': '\\b(0[xX]\\h(?>_?\\h)*|\\d(?>_?\\d)*(\\.(?![^[:space:][:digit:]])(?>_?\\d)*)?([eE][-+]?\\d(?>_?\\d)*)?|0[bB][01]+)\\b'
+    'match': '''(?x)
+      \\b(
+        \\d(?>_?\\d)*                               # 100_000
+        (\\.(?![^[:space:][:digit:]])(?>_?\\d)*)?   # fractional part
+        ([eE][-+]?\\d(?>_?\\d)*)?                   # 1.23e-4
+      | 0[xX]\\h(?>_?\\h)*
```

Comment:
If you want to do it like that then you can put the `|` on the end of the line, similar to https://github.com/atom/language-javascript/blob/74ff62f9a93de2c1976432feccce2d8ee23aa740/grammars/javascript.cson#L1091